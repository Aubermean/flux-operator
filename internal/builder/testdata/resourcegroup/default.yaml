apiVersion: fluxcd.controlplane.io/v1
kind: ResourceGroup
metadata:
  name: app1
  namespace: apps
spec:
  inputs:
    - tenant: team1
      domain: t1.example.com
      semver: ">=1.0.0-rc.0"
      semverFilter: ".*-rc.*"
      replicas: "2"
      layerSelector: |
        mediaType: "application/vnd.cncf.helm.chart.content.v1.tar+gzip"
        operation: copy
    - tenant: team2
      domain: t2.example.com
      semver: ">=1.0.0"
      replicas: "3"
  resources:
    - apiVersion: source.toolkit.fluxcd.io/v1beta2
      kind: OCIRepository
      metadata:
        name: << inputs.tenant >>-app1
        namespace: apps
      spec:
        interval: 10m
        url: oci://ghcr.io/org/charts/app1
        layerSelector: << inputs.layerSelector | default "~" | nindent 4 >>
        ref:
          semver: << inputs.semver | quote >>
          semverFilter: << inputs.semverFilter | default "*" | quote >>
    - apiVersion: helm.toolkit.fluxcd.io/v2
      kind: HelmRelease
      metadata:
        name: << inputs.tenant >>-app1
        namespace: apps
        labels:
          replicas: << inputs.replicas | quote >>
      spec:
        interval: 1h
        releaseName: << inputs.tenant >>-app1
        chartRef:
          kind: OCIRepository
          name: << inputs.tenant >>-app1
        values:
          domain: << inputs.domain >>
          replicas: << inputs.replicas | int >>
