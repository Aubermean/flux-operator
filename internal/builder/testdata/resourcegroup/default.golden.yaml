apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: OCIRepository
metadata:
  name: team1-app1
  namespace: apps
spec:
  interval: 10m
  layerSelector:
    mediaType: application/vnd.cncf.helm.chart.content.v1.tar+gzip
    operation: copy
  ref:
    semver: '>=1.0.0-rc.0'
    semverFilter: .*-rc.*
  url: oci://ghcr.io/org/charts/app1
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: OCIRepository
metadata:
  name: team2-app1
  namespace: apps
spec:
  interval: 10m
  layerSelector: null
  ref:
    semver: '>=1.0.0'
    semverFilter: '*'
  url: oci://ghcr.io/org/charts/app1
---
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  labels:
    replicas: "2"
  name: team1-app1
  namespace: apps
spec:
  chartRef:
    kind: OCIRepository
    name: team1-app1
  interval: 1h
  releaseName: team1-app1
  values:
    domain: t1.example.com
    replicas: 2
---
apiVersion: helm.toolkit.fluxcd.io/v2
kind: HelmRelease
metadata:
  labels:
    replicas: "3"
  name: team2-app1
  namespace: apps
spec:
  chartRef:
    kind: OCIRepository
    name: team2-app1
  interval: 1h
  releaseName: team2-app1
  values:
    domain: t2.example.com
    replicas: 3
---
